src/main/java/DianaAssistant.java

```diff
@@ -0,0 +1,155 @@
+        System.out.println("4. Type either mark or unmark to indicate the completion rate of your task");
+        System.out.println("5. Type list to view the your list of tasks.");
+        System.out.println("6. Type bye to exit the programme");
+        printEnclosure();
+    }
+
+    private void printEnclosure() {
+        System.out.println(ENCLOSURE);
+    }
+
+    private void printList (List<Task> list) {
+        printEnclosure();
+
+        for (int i = 0; i < list.size(); i++) {
+            Task task = list.get(i);
+            System.out.println((i + 1) + ". " + task.toString());
+        }
+        printEnclosure();
+        return;
+    }
+
+    private void toMark (List<Task> list, String input, boolean toMark) {
+        try {
+            int TaskNum = Integer.valueOf(input.substring(input.indexOf(" ") + 1)) - 1;
+            if (TaskNum >= 0 && TaskNum < list.size()) {
+                Task task = list.get(TaskNum);
+                if (toMark) {
+                    task.markAsDone();
+                    printEnclosure();
+                    System.out.println("Nice! I've marked this task as done\n" + task.toString());
+                    printEnclosure();
+                }
+                else {
+                    task.markAsNotDone();
+                    printEnclosure();
+                    System.out.println("Okay, I've marked this task as not done yet\n" + task.toString());
+                    printEnclosure();
+                }
+            }
+            else {
+                System.out.println("Please enter a number between 1 and " + (list.size() - 1));
+            }
+        }
+        catch (NumberFormatException e) {
+            System.out.println("Please enter a number between 1 and " + (list.size() - 1));
+        }
+    }
+
+    private void printAddedTask (String description, List<Task> list) {
+        printEnclosure();
+        System.out.println("Got it! I've added this task");
+        System.out.println(description);
+        System.out.println("Now you have " + list.size() + " tasks in the list");
+        printList(list);
+    }
+
+    private void addTodo (String input, List<Task> list) {
+        String description = input.substring("todo".length()).trim();
+        Todo todo = new Todo(description);
+        list.add(todo);
+        printAddedTask(description, list);
+    }
+
+    private void addDeadline (String input, List<Task> list) {
+        if (!input.contains("/by")) {
+            System.out.println("Please enter a valid deadline with the correct format:");
+            System.out.println("Eg: Read book /by 5pm");
+            return;
+        }
+        String[] parts = input.split("/by", 2);
+        String description = parts[0].substring("deadline".length()).trim();
+        String by = parts[1].trim(); // trim removes white space
+        Task deadline = new Deadline(description, by);
+        list.add(deadline);
+        printAddedTask(description, list);
```

Comment:
Could you switch the conditions in this block of code, so that the happy path (that is, the code executed when "by" is present), comes first?